src/doc/contrib/src/design.md

```diff
@@ -77,6 +77,13 @@ data in Cargo's home directory) should not *prevent* older versions from
 running, but they may cause older versions to recreate the cache, which may
 result in a performance impact.
 
+Changes to emitting hard errors, unless for fixing critical bugs, should
```

Comment:
Following the existing wording practice in this section makes this hard to read.  This makes it sound like we are editing existing hard errors, like there message.